include/dsn/utility/binary_reader.h

```diff
@@ -39,21 +40,27 @@ class binary_reader
     int read(/*out*/ bool &val) { return read_pod(val); }
 
     int read(/*out*/ std::string &s);
-    int read(char *buffer, int sz);
+    virtual int read(char *buffer, int sz);
```

Comment:
It's an important path on write, add a virtual function will add a performance cost in c++, is this change necessary？

Reply:
The performance cost introduced by vtable can be negligible. And this pull request is used for dealing with dirty mutation log.